### DISCUSSION TITLE
How to enable query logging in bob

### DISCUSSION THREAD

**Discussion by iggydv** [stephenafamo/bob#299] [Q&A] [answered]
I'm trying to figure out if it is possible to log all database queries, I haven't found any documentation regarding bob & logging. Is this something that is supported? if not, how could I get around this limitation?

**Comment by stephenafamo** [comment]
You can wrap your executor with a debug executor (or your own implementation):

```go
exec = bob.Debug(exec)
```

There is also `DebugToWriter` and `DebugToPrinter` which gives more control (compared to just writing to stdout)

**Comment by daddz** [comment, marked as answer]
You could also use a `database/sql` wrapper like https://github.com/simukti/sqldb-logger

```go
sqlDB = sqldblogger.OpenDriver(...)
db := bob.NewDB(sqlDB)
```

or if you use a different driver it may have something in store for you like e.g. https://github.com/jackc/pgx with https://pkg.go.dev/github.com/jackc/pgx/v5#hdr-Tracing_and_Logging
